src/test/java/seedu/address/model/person/PhoneContainsDigitsPredicateTest.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.PersonBuilder;
+
+public class PhoneContainsDigitsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("94351253");
+        List<String> secondPredicateKeywordList = Arrays.asList("94351253", "98765432");
+
+        PhoneContainsDigitsPredicate firstPredicate = new PhoneContainsDigitsPredicate(firstPredicateKeywordList);
+        PhoneContainsDigitsPredicate secondPredicate = new PhoneContainsDigitsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertEquals(firstPredicate, firstPredicate);
+
+        // same values -> returns true
+        PhoneContainsDigitsPredicate firstPredicateCopy = new PhoneContainsDigitsPredicate(firstPredicateKeywordList);
+        assertEquals(firstPredicate, firstPredicateCopy);
+
+        // different types -> returns false
+        assertNotEquals(1, firstPredicate);
+
+        // null -> returns false
+        assertNotEquals(null, firstPredicate);
```

Comment:
I think this should be updated too. It doesn't fail test cases probably because the `.equals(null)` is somehow invoked in some other tests.